### PR TITLE
Added local root peer provider test

### PR DESCRIPTION
**Pull request by bolt12**

### CODE REVIEW

File: ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/RootPeersDNS.hs [hunk 1]
```diff
@@ -379,6 +380,48 @@ selectPublicRootResultEvents trace = [ (t, (domain, map fst r))
...
+        localGroupEventsAddresses :: (a, Seq (Int, Map SockAddr PeerAdvertise))
+                                  -> Set SockAddr
+        localGroupEventsAddresses (_, s) =
+            Set.fromList
+          $ concatMap (Map.keys . snd)
+          $ toList s
+
+     in property
+      $ all (\x -> Set.isSubsetOf (localRootAddresses localRoots)
+                                  (localGroupEventsAddresses x)
```
Comment (coot) [inline ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/RootPeersDNS.hs:420]:
```suggestion
      $ all (\x -> localRootAddresses localRoots
                   `Set.isSubsetOf`
                   localGroupEventsAddresses x
```

File: ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/RootPeersDNS.hs [hunk 2]
```diff
@@ -379,6 +380,48 @@ selectPublicRootResultEvents trace = [ (t, (domain, map fst r))
...
+        -- local root addresses
+        localRootAddresses :: [(a, Map RelayAccessPoint PeerAdvertise)]
+                           -> Set SockAddr
+        localRootAddresses lrp =
+          Set.fromList
+          [ toSockAddr (ip, port)
+          | (_, m) <- lrp
+          , RelayAccessAddress ip port <- Map.keys m
+          ]
+        localGroupEventsAddresses :: (a, Seq (Int, Map SockAddr PeerAdvertise))
```
Comment (coot) [inline ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/RootPeersDNS.hs:411]: Please add an empty line above.

**Review by coot — COMMENTED** [review]
LGTM, just a few minor comments.

**Review by coot — APPROVED** [review]

**Comment by bolt12** [comment]
bors merge

**Pull request merged**